src/gui/menu.py

```diff
@@ -0,0 +1,212 @@
+                                       (ButtonType.TEXT_NARROW, (ButtonId.ENTER_GAME, "PLAY"))]),
+            PageId.SETTINGS: Page(header="SETTINGS",
+                                  buttons=[(ButtonType.MULTI_CHOICE_SHORT,
+                                            (None, {ButtonId.SOUND_ON: "SOUND ON", ButtonId.SOUND_OFF: "SOUND OFF"})),
+                                           (ButtonType.TEXT_WIDE, (ButtonId.BACK, "BACK"))]),
+            PageId.SCORES: Page(header="BEST SCORES",
+                                buttons=[(ButtonType.TEXT_AREA, (ButtonId.HARD, text_area_lines)),
+                                         (ButtonType.TEXT_WIDE, (ButtonId.BACK, "BACK"))]),
+            PageId.ABOUT: Page(header="CREDITS",
+                               buttons=[(ButtonType.TEXT_AREA,
+                                         (ButtonId.HARD, ["Jakub Gros", "Sebastian Kucharzyk", "Mateusz Olejarz"])),
+                                        (ButtonType.TEXT_WIDE, (ButtonId.BACK, "BACK"))]),
+            PageId.HELP: Page(header="HELP",
+                              buttons=[(ButtonType.TEXT_AREA,
+                                        (ButtonId.HARD, ["Go left: left arrow", "Go right: right arrow",
+                                                         "Go up: up arrow", "Go down: down arrow", "Action: A button"])),
+                                       (ButtonType.TEXT_WIDE, (ButtonId.BACK, "BACK"))]),
+        }
+        self._is_moving = DotDict({"up": False, "down": False, "left": False, "right": False, "into": False})
+        self._settings = Settings()
+        self._active_page_id = PageId.DEFAULT
+        self._is_running = True
+        self._chosen_game = None
+
+    def process_events(self, joystick):
+        for event in pygame.event.get():
+            # TODO sk: remove these lines when game ready
+            if event.type == QUIT or (event.type == KEYDOWN and event.key == K_ESCAPE):
+                sys.exit(0)
+
+            if event.type in {JOYBUTTONUP, JOYBUTTONDOWN, JOYAXISMOTION}:
+                joystick.process_event(event)
+                if joystick.is_a_pressed():
+                    self.get_into()
+                if joystick.is_arrow_downdir_pressed():
+                    self.move_down()
+                if joystick.is_arrow_updir_pressed():
+                    self.move_up()
+                if joystick.is_arrow_rightdir_pressed():
+                    self.move_right()
+                if joystick.is_arrow_leftdir_pressed():
+                    self.move_left()
+
+    def is_running(self):
+        return self._is_running
+
+    def move_up(self):
+        self._is_moving.up = True
+
+    def move_down(self):
+        self._is_moving.down = True
+
+    def move_left(self):
+        self._is_moving.left = True
+
+    def move_right(self):
+        self._is_moving.right = True
+
+    def get_into(self):
+        self._is_moving.into = True
+
+    def update(self):
```

Comment:
I can't think of any improvement here. Could you tell me where and how you would see that?

Reply:
For sure I would add functions for processing each case in if statements so that it was easier to see all of the paths that are possible. E.g. 

instead of

```
  elif self._is_moving.down:
            self._pages[self._active_page_id].move_down()
            if self._settings.sound_on:
                SoundRsc.button_switch.play()
            self._is_moving.down = False
```

I would write

```
  elif self._is_moving.down:
        processMovingDown()
```

Reply:
Moreover, doing this kind of refactoring you will find that some functions are similar, for example move_left and move_right, move_up and move_down

Reply:
Ok